### PR TITLE
don't use map output tracker in localDiskShuffleReadSupport

### PR DESCRIPTION
**Pull request by yifeih**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/shuffle/io/LocalDiskShuffleReadSupport.scala [hunk 1]
```diff
@@ -80,23 +73,26 @@ private class ShuffleBlockFetcherIterable(
...
-          })),
+      blockMetadata.asScala
+        .groupBy(shuffleBlockInfo => shuffleBlockInfo.getShuffleLocation.get())
+        .map{ case (blockManagerId, shuffleBlockInfos) =>
+          (blockManagerId,
+            shuffleBlockInfos.map(info =>
+              (ShuffleBlockId(
+                info.getShuffleId,
+                info.getMapId,
+                info.getReduceId).asInstanceOf[BlockId],
```
Comment (mccheah) [inline core/src/main/scala/org/apache/spark/shuffle/io/LocalDiskShuffleReadSupport.scala]: Do we really have to do `asInstanceOf[BlockId]` here?

**Comment by mccheah** [comment]
Double check that there's test coverage . for this - but LGTM

File: core/src/main/scala/org/apache/spark/shuffle/io/LocalDiskShuffleReadSupport.scala (hunk 1 above)
Comment (yifeih) [reply core/src/main/scala/org/apache/spark/shuffle/io/LocalDiskShuffleReadSupport.scala]: Yea otherwise Scala complains that it's the wrong type :/

Comment (sritchie) [reply core/src/main/scala/org/apache/spark/shuffle/io/LocalDiskShuffleReadSupport.scala:92]: You should be able to add types to `map` (as in `map[inputType, BlockId]`) to remove the cast!

**Pull request merged**